### PR TITLE
fix: detect non-interactive stdio and use "normal" IO operations

### LINKED ISSUE #199: stacks rm: panic: runtime error: invalid memory address or nil pointer dereference
```
influx version
Influx CLI dev (git: none) build_date: 2021-07-20T14:47:49Z

yes | influx stacks rm --stack-id $STACK_ID -c docker --skip-verify
ID			OrgID			Active	Name	Description	Num Resources	Sources		URLs	Created At				Updated At
07dd0f69dbb02000	ebbd2fe8e567f379	true				4		[byte stream]	[]	2021-07-20 14:09:43.7901653 +0000 UTC	2021-07-20 14:09:43.8538043 +0000 UTC
? Confirm removal of the stack[07dd0f69dbb02000] and all associated resources (y/N) ^[[31;204R^[[31;85R
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x1436487]

goroutine 1 [running]:
github.com/AlecAivazis/survey/v2/terminal.(*RuneReader).ReadLine(0xc0005546c0, 0x0, 0x1ca8cc8, 0xc00000e018, 0x18ebd80, 0xc00000e020, 0xc0005546c0)
	/Users/rsavage/go/pkg/mod/github.com/!alec!aivazis/survey/v2@v2.2.9/terminal/runereader.go:56 +0x447
github.com/AlecAivazis/survey/v2.(*Confirm).getBool(0xc0000fa2c0, 0x1834400, 0xc0000b4468, 0x178ad00, 0x0, 0x0)
	/Users/rsavage/go/pkg/mod/github.com/!alec!aivazis/survey/v2@v2.2.9/confirm.go:57 +0x16e
github.com/AlecAivazis/survey/v2.(*Confirm).Prompt(0xc0000fa2c0, 0xc0000b4468, 0xc00000e010, 0x18f21d8, 0xc00000e018, 0x18ebd80)
	/Users/rsavage/go/pkg/mod/github.com/!alec!aivazis/survey/v2@v2.2.9/confirm.go:136 +0x16d
github.com/AlecAivazis/survey/v2.Ask(0xc00032a5f0, 0x1, 0x1, 0x16e48a0, 0xc000029190, 0xc00032a5e8, 0x1, 0x1, 0xc0000f8750, 0xc00032a628)
	/Users/rsavage/go/pkg/mod/github.com/!alec!aivazis/survey/v2@v2.2.9/survey.go:293 +0x57b
github.com/AlecAivazis/survey/v2.AskOne(...)
	/Users/rsavage/go/pkg/mod/github.com/!alec!aivazis/survey/v2@v2.2.9/survey.go:236
github.com/influxdata/influx-cli/v2/pkg/stdio.(*terminalStdio).GetConfirm(0x1d85980, 0xc0002ba050, 0x4b, 0x1)
	/Users/rsavage/go/src/github.com/influxdata/influx-cli/pkg/stdio/console.go:110 +0x1a5
github.com/influxdata/influx-cli/v2/clients/stacks.Client.Remove(0x18fc1d8, 0x1d85980, 0x0, 0xc0000c34a6, 0x6, 0xc0000aa420, 0x16, 0xc000098b20, 0x7, 0xc000098b50, ...)
	/Users/rsavage/go/src/github.com/influxdata/influx-cli/clients/stacks/stacks.go:142 +0xa14
main.newStacksRemoveCmd.func1(0xc0000b4580, 0x2, 0xc0001cad40)
	/Users/rsavage/go/src/github.com/influxdata/influx-cli/cmd/influx/stacks.go:187 +0x318
github.com/urfave/cli.HandleAction(0x17138a0, 0xc0001cad50, 0xc0000b4580, 0xc0000b4580, 0x0)
	/Users/rsavage/go/pkg/mod/github.com/urfave/cli@v1.22.5/app.go:524 +0x105
github.com/urfave/cli.Command.Run(0x17f391c, 0x2, 0x0, 0x0, 0xc000204680, 0x2, 0x2, 0x181f78d, 0x2e, 0x0, ...)
	/Users/rsavage/go/pkg/mod/github.com/urfave/cli@v1.22.5/command.go:173 +0x579
github.com/urfave/cli.(*App).RunAsSubcommand(0xc00009a000, 0xc0001389a0, 0x0, 0x0)
	/Users/rsavage/go/pkg/mod/github.com/urfave/cli@v1.22.5/app.go:405 +0x914
github.com/urfave/cli.Command.startApp(0x17f6cfb, 0x6, 0x0, 0x0, 0x0, 0x0, 0x0, 0x182b208, 0x42, 0x0, ...)
	/Users/rsavage/go/pkg/mod/github.com/urfave/cli@v1.22.5/command.go:372 +0x7ff
github.com/urfave/cli.Command.Run(0x17f6cfb, 0x6, 0x0, 0x0, 0x0, 0x0, 0x0, 0x182b208, 0x42, 0x0, ...)
	/Users/rsavage/go/pkg/mod/github.com/urfave/cli@v1.22.5/command.go:102 +0x9d4
github.com/urfave/cli.(*App).Run(0x1ee7420, 0xc00001e080, 0x8, 0x8, 0x0, 0x0)
	/Users/rsavage/go/pkg/mod/github.com/urfave/cli@v1.22.5/app.go:277 +0x808
main.main()
	/Users/rsavage/go/src/github.com/influxdata/influx-cli/cmd/influx/main.go:61 +0x55
```

### PR DESCRIPTION
**Pull request by danxmoran**
Closes #199 

Also works for commands other than `stacks`:
Screen Shot 2021-07-21 at 4 34 00 PM
Could be useful for e2e tests

### CODE REVIEW

File: pkg/stdio/stdio_internal_test.go [hunk 1]
```diff
@@ -0,0 +1,284 @@
...
+func TestTerminalStdIO_Banner(t *testing.T) {
+	t.Parallel()
+
+	tmp, err := os.MkdirTemp("", "")
+	require.NoError(t, err)
+	defer os.RemoveAll(tmp)
+
+	stdin, err := os.Create(filepath.Join(tmp, "stdin"))
+	require.NoError(t, err)
+	defer stdin.Close()
```
Comment (davidby-influx) [inline pkg/stdio/stdio_internal_test.go:21]: Non-blocking: is it worth checking the return from the Close() calls?

**Review by davidby-influx — APPROVED** [review]
Non-blocking - I feel like there is an opportunity for another layer of interfaces or maybe inheritance here because of the similarity of Write() and WriteErr() implementations. Might be overkill, though, with unnecessary abstraction.

Overall, LGTM. All comments non-blocking.

File: pkg/stdio/stdio_internal_test.go (hunk 1 above)
Comment (danxmoran) [reply pkg/stdio/stdio_internal_test.go:21]: I don't typically check them (especially in test code), I figure if `Close()` fails then some other catastrophic thing must have happened that will show in the test output

**Comment by danxmoran** [comment]
Agreed that it feels like there's room for another layer. I spent an hour or two trying to tease it apart but everything I came up with felt like overkill, so I gave up and went with this version. Maybe it'll emerge in future refactorings.

**Pull request merged**